### ISSUE TITLE
[linux] Investigate gdb hang while running `make run-test` + WITH_OSMESA=1

### ISSUE THREAD

**Issue by brunoabinader** [mapbox/mapbox-gl-native#6918]
After #6899 gets merged, we should be able to not only run Node render+query tests, but also regular unit tests that requires rendering using OSMesa backend instead of GLX.

However, currently Travis bot _hangs_ when running `make tests` via gdb when using the OSMesa backend:
> gdb -batch -return-child-result -ex 'set print thread-events off' -ex 'run' -ex 'thread apply all bt' --args build/linux-x86_64/Debug/mbgl-test --gtest_catch_exceptions=0 --gtest_filter=*
> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
> The build has been terminated

**Comment by jfirebaugh** [comment]
No longer relevant; we removed testing against OSMesa in #7787.
